src/Microsoft.CodeAnalysis.Analyzers/Core/CodeAnalysisDiagnosticsResources.resx

```diff
@@ -529,4 +529,21 @@
   <data name="PreferIsKindFix" xml:space="preserve">
     <value>Use 'IsKind' instead of 'Kind'</value>
   </data>
+  <data name="MissingSourceGeneratorAttributeDescription" xml:space="preserve">
+    <value>'{0}' implements 'ISourceGenerator' but does not have the 'Generator' attribute.</value>
+  </data>
+  <data name="MissingSourceGeneratorAttributeMessage" xml:space="preserve">
+    <value>Missing 'Generator' attribute</value>
+  </data>
+  <data name="MissingSourceGeneratorAttributeTitle" xml:space="preserve">
+    <value>Missing 'Generator' Attribute</value>
+  </data>
+  <data name="AddGeneratorAttribute_1" xml:space="preserve">
+    <value>Apply Generator attribute for '{0}'.</value>
+    <comment>"Generator" is a named type and should not be changed</comment>
```

Comment:
There is a special syntax for locking localization. e.g:

https://github.com/dotnet/roslyn-analyzers/blob/6f7420ab11fb2fb639fa2242657885c8a614f97e/src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticsAnalyzersResources.resx#L380

Reply:
Interesting. I've always just left a comment like this and translators worked it out. Wonder what benefit `{Locked="Shared"}` has

Reply:
@ryzngard I have no idea of the internal localization process 😄

Reply:
I'm going to leave now and follow up tomorrow for what's going on. Likely you're right, but I want to know what the difference is.

Reply:
I have seen the same pattern described by @Youssef1313 on this repo.

Reply:
I'll switch to locked here